### PR TITLE
Fix issue #75 + minor improvements

### PR DESCRIPTION
**Pull request by playdohface**
- `Storage` now returns `Option<Message>` for `.get(id)` to handle absence of values correctly and its implementors have been adjusted accordingly
- remove uses of `unwrap()` in adjacent code
- expand tests for `Storage` implementations

### CODE REVIEW

**Comment by stappersg** [comment]
`Fix issue #75 + minor improvements` please make that two separate merge requests.

(1 commit should change 1 thing only)

**Comment by playdohface** [comment]
> `Fix issue #75 + minor improvements` please make that two separate merge requests.
> 
> (1 commit should change 1 thing only)

Where would you suggest to split it?

**Comment by stappersg** [comment]
On Sat, Aug 10, 2024 at 02:01:11PM -0700, playdohface wrote:
> On Sat, Aug 10, 2024, stappersg wrote:
> > `Fix issue #75 + minor improvements` please make that two separate merge requests.
> } `Fix issue #75 + minor improvements` please make that at least two separate merge requests.
> > 
> > (1 commit should change 1 thing only)
> 
> Where would you suggest to split it? 
> 

Split at the plus sign.   8^)

Fix just the crash on non-existing Message-ID request.


And split the minor improvements also in separate commits.




I did see that the merge request is from branch 'main'.
That might explain the "Where would you suggest to split it?"
I think that my "please make separate merge requests" is better
expressed as "use git branches".


Regards
Geert Stappers
-- 
Silence is hard to parse

**Comment by playdohface** [comment]
Did you read the code, or reproduce the issue? There is the change in the trait, and the refactoring of its implementors and error handling in the endpoints (which amounts to fixing the issue) and tests asserting the expected behaviour. Neither of which makes sense without the others. 
It's a small and atomic change. Why (and again: where, in the code) would it be beneficial to artificially split? To what end?
Same for the name of the fork branch. If there is a reason or specific request, I'll change it, but I don't see that here. 
A good enough reason for any change would be: simple preference of an actual maintainer. But to cater to taste, we need specific instructions.

**Comment by stappersg** [comment]
My approach is

```diff
--- a/mailtutan-lib/src/storage/memdir.rs
+++ b/mailtutan-lib/src/storage/memdir.rs
@@ -80,7 +80,10 @@ impl Storage for Memdir {
     fn get(&self, item: usize) -> Message {
         let filename = format!("{}.eml", item);
 
-        let bytes = fs::read(self.path.join(filename)).unwrap();
+        let bytes = match fs::read(self.path.join(filename)) {
+            Ok(bytes) => bytes,
+            Err(_) => b"message not found".to_vec(),
+        };
 
         let mut message = Message::from(&bytes).unwrap();
         message.id = Some(item);
```

a.k.a.  https://github.com/mailtutan/mailtutan/pull/78/files
